Examples/S3EventNotifier/Sources/main.swift

```diff
@@ -0,0 +1,86 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftAWSLambdaRuntime open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftAWSLambdaRuntime project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import AWSLambdaEvents
+import AWSLambdaRuntime
+import AsyncHTTPClient
+
+#if canImport(FoundationEssentials)
+import FoundationEssentials
+#else
+import Foundation
+#endif
+
+let httpClient = HTTPClient.shared
+
+enum LambdaError: Error {
+    case noNotificationRecord
+    case missingEnvVar(name: String)
+
+    var description: String {
+        switch self {
+        case .noNotificationRecord:
+            "No notification record in S3 event"
+        case .missingEnvVar(let name):
+            "Missing env var named \(name)"
+        }
+    }
+}
+
+let runtime = LambdaRuntime { (event: S3Event, context: LambdaContext) async throws -> APIGatewayV2Response in
+    do {
+        context.logger.debug("Received S3 event: \(event)")
+
+        guard let s3NotificationRecord = event.records.first else {
+            throw LambdaError.noNotificationRecord
+        }
+
+        let bucket = s3NotificationRecord.s3.bucket.name
+        let key = s3NotificationRecord.s3.object.key.replacingOccurrences(of: "+", with: " ")
+
+        guard let apiURL = ProcessInfo.processInfo.environment["API_URL"] else {
+            throw LambdaError.missingEnvVar(name: "API_URL")
+        }
+
+        let body = """
+            {
+                "bucket": "\(bucket)",
+                "key": "\(key)"
+            }
+            """
+
+        context.logger.debug("Sending request to \(apiURL) with body \(body)")
```

Comment:
I would simplify and not send a POST request to an API.
That obliges developers willing to test this function to setup an HTTPS API endpoint, which is unnecessary to understand this example.

I propose to just log relevant informations from the event.  The specific handling of these events are application specific, we can't assume a dev will have to call a REST API. I think it just adds complexity (and maybe confusion) to the example.

You can add comments to suggest what can be done after receiving the event